app/controllers/verify/finance_controller.rb

```diff
@@ -25,6 +32,30 @@ def render_form
       end
     end
 
+    def confirm_finance_with_vendor?
+      session_id = idv_session.resolution.session_id
+      idv_session.financials_confirmation = idv_agent.submit_financials(financials, session_id)
+      track_idv_event
+      idv_session.financials_confirmation.success?
+    end
+
+    def financials
+      finance_type = idv_finance_form.finance_type
+      { finance_type => idv_finance_form.idv_params[finance_type] }
+    end
+
+    def financials_confirmation_failed?
+      idv_finance_form.finance_type && !idv_session.financials_confirmation.try(:success?)
+    end
```

Comment:
If I understand this correctly, you are treating the `Idv::FinanceForm` validations separately from the Vendor API validation, and you are creating this helper method for use in the view in the scenario where the Vendor validation fails, in order to display an error message on the page.

What do you think about keeping all of the form submission logic in one place, in the `Idv::FinanceForm`? This would allow us to track both the form errors and vendor errors in one analytics call, and would not necessitate this helper method, or any additional view work. We would only need to add an appropriate error message, which will automatically be displayed next to the field that failed the validation.